services/core-web/src/components/common/DocumentTable.js

```diff
@@ -47,11 +57,55 @@ const defaultProps = {
   additionalColumnProps: [],
   documentColumns: null,
   documentParent: null,
-  defaultSortKeys: ["upload_date", "dated"], // keys to sort by when page loads
+  isLoaded: false,
+  noSubTableExpandableRows: false,
+  defaultSortKeys: ["upload_date", "dated", "update_timestamp"], // keys to sort by when page loads
   view: "standard",
   canArchiveDocuments: false,
 };
 
+const renderFileLocation = (documentTypeCode) => {
+  let location = "N/A";
+  switch (documentTypeCode) {
+    case "PRM":
+      location = "Primary Document";
+      break;
+    case "SPT":
+      location = "Spatial Component";
+      break;
+    case "SPR":
+      location = "Supporting Document";
+      break;
+  }
+
+  return location;
+};
+
+const renderFileType = (file) => {
+  const index = file.lastIndexOf(".");
+  return index === -1 ? "N/A" : file.substr(index);
+};
+
+const parseFiles = (versions, documentType) =>
+  versions.map((version, index) => ({
+    key: version.mine_document_version_guid,
+    file_location: renderFileLocation(documentType) || Strings.EMPTY_FIELD,
+    file_type: renderFileType(version.document_name) || Strings.EMPTY_FIELD,
+    number_of_versions: index === 0 ? versions.length - 1 : 0,
+    ...version,
+  }));
+
+const transformRowData = (document) => {
```

Comment:
I think I see the issue-
to use the table as designed, we want each record to look something like:
document_guid: abc,
upload_date: today,
...(other attributes about that document),
oldVersions: [{version number, guid, upload date, etc}]

where the old versions are an array attribute of the document. Then we can use the table normally, with getDataSource = record => record.oldVersions, and just pass in columns for the latest version, and another array of columns for the expanded rows. The table will handle the rest without any changes to it.